mmd/compositeconn.go

```
@@ -240,9 +240,25 @@ func (c *CompositeConn) createAndInitDirectConnection(service string) (*ConnImpl
 }
 
 var env = computeEnv()
-var nameserverUrl = env + ".k8s.peak6.net:53"
-var istioIngressUrl = env + ".istioingress.peak6.net"
+var nameserverHost = getEnv("NAMESERVER_HOST", env + ".k8s.peak6.net")
+var istioIngressHost = getEnv("ISTIO_INGRESS_HOST", env + ".istioingress.peak6.net")
 var _, isInK8s = os.LookupEnv("KUBERNETES_SERVICE_HOST")
+var useIstioIngress = getEnvBool("USE_ISTIO_INGRESS", !isInK8s)
+
+func getEnv(key, fallback string) string {
+	if value, ok := os.LookupEnv(key); ok {
+		return value
+	}
+	return fallback
+}
+
+func getEnvBool(key string, fallback bool) bool {
+	if value, ok := os.LookupEnv(key); ok {
+		var result, _ = strconv.ParseBool(value)
```

maybe check if parse success here

---

What do you think the right course of action is if it's a parse failure, panic?  I used this approach (consider the value false if it's uparseable), but failing fast for invalid config probably makes more sense..

---

I think a panic is acceptable here given the environment variable has been set, but it is not parsable as a bool, we would probably want to know that right away. Nice catch Xizi.

---

Done, and moved these methods to mmd.go so that connconfig.go can also use them